packages/dds/merge-tree/src/test/mergeTreeOperationRunner.ts

```diff
@@ -29,6 +29,59 @@ export type TestOperation = (
 	random: IRandom,
 ) => IMergeTreeOp | undefined;
 
+export const insertField: TestOperation = (
+	client: TestClient,
+	opStart: number,
+	opEnd: number,
+	random: IRandom,
+) => {
+	const numberText = random.string(5, "0123456789");
+	if (
+		// start is not a number
+		Number.isNaN(Number(client.getText(opStart, opStart + 1))) &&
+		// previous is not a number
+		(opStart === 0 ||
+			opStart >= client.getLength() ||
+			Number.isNaN(Number(client.getText(opStart - 1, opStart))))
+	) {
+		return client.insertTextLocal(opStart, numberText);
+	}
+};
+
+const numberRange = (
+	client: TestClient,
+	start: number,
+): { startPos: number | undefined; endPos: number | undefined } => {
+	let startPos = start;
+	while (startPos > 0 && !Number.isNaN(Number(client.getText(startPos, startPos + 1)))) {
+		startPos--;
+	}
+	return { startPos, endPos: startPos + 4 };
+};
+
+export const obliterateField: TestOperation = (
```

Comment:
In the case we care about, the field value never only gets obliterated. It's always obliterate [with endpoint expansion] + insert updated value. (If it's just obliterating, then we start hitting the zero-length case.) Not sure how much this distinction matters here; I think it might when two clients are competing to replace the same field value.